File: algorithm/1_Two_Sum/Two_Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> hash_map;
        int nums_size = nums.size();
        int complement;
        for(int i = 0; i < nums_size; i++){
            complement = target - nums[i];
            if(hash_map.find(complement)!=hash_map.end()){
                return vector<int>({hash_map[complement],i});
            }
            hash_map[nums[i]]=i;
        }
        return vector<int>();
    }
};

File: algorithm/1_Two_Sum/Two_Sum.go
func twoSum(nums []int, target int) []int {
    mapComplement := make(map[int]int)
    for index,key := range nums{
        if _,ok := mapComplement[target-key]; ok{
            return []int{mapComplement[target-key],index}
        }
        mapComplement[key] = index
    }
    return []int{}
}